src/logfile_values_experimental.py

```diff
@@ -0,0 +1,224 @@
+import gzip
+import logging
+
+from argparse import ArgumentParser, ArgumentTypeError
+from datetime import timedelta
+
+
+class Metric:
+    def __init__(self, arg):
+        if ':' not in arg:
+            raise ArgumentTypeError('Argument must have ":"')
+        parts = arg.split(':')
+        if len(parts) != 4 and len(parts) != 2:
+            raise ValueError('Wrong number of options')
+
+        # Accepts 2 or 4 options. If 2 then function and period are NULL
+        parts += [None] * (4 - len(parts))
+        self.name, column, function, period = parts
+        if period:
+            pattern = re.compile("^\d+[A-Za-z]$")
+            if not pattern.match(period):
+                raise ArgumentTypeError('Period must have number and unit')
+        if not column.isdecimal():
+            raise ArgumentTypeError('Column must be a number')
+        self.column = int(column)
+        self.function = function
+        self.period = period
+        # Here is container for metric values
+        self.values = []
+        self.last_value = 0
+        self.now = int(time.time())
+
+    def get_timeshift(self):
+        if self.period:
+            value = int(self.period[:-1])
+            unit = self.period[-1].lower()
+            if unit == 's':
+                return timedelta(seconds=value).total_seconds()
+            if unit == 'm':
+                return timedelta(minutes=value).total_seconds()
+            if unit == 'h':
+                return timedelta(hours=value).total_seconds()
+            if unit == 'd':
+                return timedelta(days=value).total_seconds()
+        else:
+            return 0
+
+    def get_median(self):
+        sorted_list = sorted(self.values)
+        i = len(sorted_list)
+        if not i % 2:
+            return (sorted_list[(i // 2) - 1] + sorted_list[i // 2]) / 2
+        return sorted_list[i // 2]
+
+    def get_sum(self):
+        return sum(self.values)
+
+    def get_count(self, v=0):
+        return sum(1 for x in self.values if x > v)
+
+    def get_mean(self):
+        return sum(self.values) / len(self.values)
+
+    def get_min(self):
+        return min(self.values)
+
+    def get_max(self):
+        return max(self.values)
+
+    def get_last_value(self):
+        return self.last_value
+
+    def get_metric_value(self):
+        if 'count_' in self.function:
+            return int(self.get_count(int(self.function.split('_')[1])))
+        return int(getattr(self, 'get_' + self.function)())
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='logfile_values')
+    parser.add_argument('--file', default='/var/log/messages')
+    parser.add_argument('--metric', type=Metric, nargs='+')
+    parser.add_argument('--time_format', default='%Y-%m-%dT%H:%M:%S')
+    parser.add_argument('--arch', action='store_true')
+    parser.add_argument('--debug', '-d', action='store_true')
+    parser.add_argument('--timezone', default='Europe/Berlin')
```

Comment:
We can also call this `--time-zone` is it is written separated in English.  `UTC` would make a better default.